lib/ibandit/local_details_cleaner.rb

```diff
@@ -441,20 +441,19 @@ def self.clean_ro_details(local_details)
     end
 
     def self.clean_se_details(local_details)
-      converted_details = SwedishDetailsConverter.new(
-        branch_code: local_details[:branch_code],
-        account_number: local_details[:account_number]
-      ).convert
-
-      bank_code = local_details[:bank_code] ||
-                  converted_details[:swift_bank_code]
-
-      {
-        account_number:       converted_details[:account_number],
-        branch_code:          converted_details[:branch_code],
-        swift_bank_code:      bank_code,
-        swift_account_number: converted_details[:swift_account_number]
-      }
+      if local_details[:bank_code]
+        # If a bank_code was provided without a branch code we're (probably)
+        # dealing with SWIFT details and should just return them.
+        return local_details.merge(
+          swift_account_number: local_details[:account_number],
+          swift_bank_code: local_details[:bank_code]
+        )
```

Comment:
Is merging right here? If we think we've been given SWIFT details, then we probably shouldn't return any local details, as we can't deduce those from the SWIFT.